core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -100,8 +101,8 @@ object ConsoleConsumer extends Logging {
         consumer.receive()
       } catch {
         case e: Throwable => {
-          error("Error processing message, stopping consumer: ", e)
-          consumer.stop()
+          error("Error processing message, terminating consumer process: ", e)
+          // Consumer will be stopped in shutdown hook
```

Comment:
Yeah you are right, and looking into the code again I think consumer.close() is still needed in the exception handling case since for the new consumer it actually just wakes up the consumer from polling, otherwise the later consumer.cleanup() which in new consumer really calls the shutdown function will be blocked. @rajinisivaram thoughts?